File: frontend/src/services/aiService.js
import api from "./api";

export const analyzeResume = async (resumeText) => {
    const response = await api.post(
        "/ai/resume/analyze",
        resumeText,
        {
            headers: {
                "Content-Type": "text/plain",
            },
        }
    );

    return response.data;
};

export const generateInterviewQuestions = async (
    topic,
    difficulty,
    count
) => {
    const response = await api.post(
        "/ai/interview/questions",
        null,
        {
            params: {
                topic,
                difficulty,
                count,
            },
        }
    );

    return response.data;
};

export const evaluateAnswer = async (
    question,
    answer
) => {
    const response = await api.post(
        "/ai/interview/evaluate",
        null,
        {
            params: {
                question,
                answer,
            },
        }
    );

    return response.data;
};

File: backend/src/main/java/com/prepbuddy/service/AIService.java
package com.prepbuddy.service;

import com.prepbuddy.dto.ResumeAnalysisResponse;
import com.prepbuddy.dto.AnswerEvaluationResponse;
import java.util.List;

public interface AIService {
    String generateResponse(String prompt);
    ResumeAnalysisResponse analyzeResume(String resumeText);
    List<String> generateInterviewQuestions(
            String topic,
            String difficulty,
            int count
    );
    AnswerEvaluationResponse evaluateAnswer(
            String question,
            String answer
    );
}
